### ISSUE TITLE
[Bug]: Unexpected spaces and cursor positioning in custom prompt

### ISSUE THREAD

**Issue by DavisRayM** [MrPicklePinosaur/shrs#285]
## What went wrong?

Whenever a new line char is used within the `styled!` macro it ends up adding some additional spaces(attached image) to the prompt while the cursor is placed at the beginning of the new line.

Using this custom prompt left function:
```rs
let git_branch = line_ctx
    .ctx
    .state
    .get::<DirParseState>()
    .and_then(|s| s.get_module_metadata::<Git>("git"))
    .map(|g| format!(" {}", g.branch));

if let Some(branch) = git_branch {
    styled!(@(blue, bold)top_pwd(), " on ", @(blue, bold)branch, "\n", "> ")
} else {
    styled!(@(blue, bold)top_pwd(), " > ")
}
```

I was expecting the prompt to look like:

```
<dir> on <branch_info>
>
```

![Screenshot from 2023-12-15 00-47-11](https://github.com/MrPicklePinosaur/shrs/assets/25849009/e5038fd9-c62b-45b5-9402-7bdfb076e253)

**Comment by MrPicklePinosaur** [comment]
I believe that prompt has no concept of newlines currently. Will look into this

**Comment by nithinmuthukumar** [comment]
Resolved by #287
